modvendor/download/report.go

```diff
@@ -109,6 +109,41 @@ func (r Report) Verbose() string {
 	return strings.Join(report, "\n")
 }
 
+// RemoveIgnoredByKind removes all ignored from this report that have errors
+// with the given kind.
+func (r *Report) RemoveIgnoredByKind(kind errors.Kind) {
+	ignored := []IgnoredVendor{}
+	for _, v := range r.Ignored {
+		if !errors.IsKind(v.Reason, kind) {
+			ignored = append(ignored, v)
+		}
+	}
+	r.Ignored = ignored
+}
+
+// IsEmpty returns true if the report is empty (nothing to report).
+func (r Report) IsEmpty() bool {
+	return len(r.Vendored) == 0 &&
+		len(r.Ignored) == 0 &&
+		r.Error == nil
+}
+
+func (r *Report) merge(other Report) {
+	if other.Error != nil {
+		if r.Error == nil {
+			r.Error = other.Error
+		} else {
+			r.Error = errors.L(r.Error, other.Error)
+		}
+	}
+
+	for k, v := range other.Vendored {
+		r.Vendored[k] = v
```

Comment:
merging the reports only keep the last report entries, is this ok? sorry if it's a stupid question but I don't remember the details in this case. Can you explain?

Reply:
I had to define some semantics for this, but in practice it would not happen today since when we vendor something with success any attempt of vendoring it again will result in a IgnoredVendor. But changes on the code could make this happen and right now we just override indeed, to avoid that we would require some more changes on the report to allow duplicated entires (the current map design doesn't work then).

Maybe this could be addressed later, really not sure what to do here (just saw it was safe for now to do overriding).